File: c/questions/insert.c
int** insert(int** intervals, int intervalsSize, int* intervalsColSize, int* newInterval, int newIntervalSize, int* returnSize, int** returnColumnSizes) {
  int **ans = (int **)malloc(sizeof(int *) * (intervalsSize+1));
  *returnColumnSizes = (int*)malloc(sizeof(int) * (intervalsSize+1));
  int index = 0;
  int i;
  for (i=0; i<intervalsSize+1; i++) {
    (*returnColumnSizes)[i] = 2;
  }
  i = 0;
  while (i<intervalsSize && intervals[i][1]<newInterval[0]) {
    ans[index++] = intervals[i];
    i++;
  }
  if (i<intervalsSize) {
    newInterval[0] = fmin(newInterval[0], intervals[i][0]);
  }
  while (i<intervalsSize && intervals[i][0]<=newInterval[1]) {
    newInterval[1] = fmax(newInterval[1], intervals[i][1]);
    i++;
  }
  ans[index++] = newInterval;
  while (i<intervalsSize) {
    ans[index++] = intervals[i];
    i++;
  }
  *returnSize = index;
  return ans;
}

File: python/questions/insert.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        s = newInterval[0]
        e = newInterval[1]
        p = 0;
        ans = [];
        while (p<len(intervals) and intervals[p][1]<s):
            ans.append(intervals[p])
            p = p + 1
        if (p < len(intervals)):
            s = min(s, intervals[p][0])
        while (p<len(intervals) and intervals[p][0]<=e):
            e = max(e, intervals[p][1])
            p = p + 1
        ans.append([s, e])
        while (p<len(intervals)):
            ans.append(intervals[p])
            p = p + 1
        return ans
